### PR TITLE
Python 2 cleanup

### PR DESCRIPTION
**Pull request by leezu**
## Description ##
This PR drops some Python 2 compatibility code from the codebase. Makes the code easier to read.

### Changes ###
- [X] Drop some python 2 compatibility handlers from mxnet python package and tests

### CODE REVIEW

File: docs/python_docs/python/tutorials/packages/gluon/text/gnmt.rst [hunk 1]
```diff
@@ -391,7 +390,7 @@ testing datasets.
 
 
     def write_sentences(sentences, file_path):
-        with io.open(file_path, 'w', encoding='utf-8') as of:
+        with open(file_path, 'w', encoding='utf-8') as of:
```
Comment (larroy) [inline docs/python_docs/python/tutorials/packages/gluon/text/gnmt.rst]: Isn't utf8 by default?

Comment (leezu) [reply docs/python_docs/python/tutorials/packages/gluon/text/gnmt.rst]: No. Default is platform dependent. https://docs.python.org/3/library/functions.html#open

File: ci/docker/runtime_functions.sh [hunk 2]
```diff
@@ -43,11 +43,10 @@ scala_prepare() {
 
 check_cython() {
     set -ex
-    local python_ver=$1
-    local is_cython_used=$(python${python_ver} <<EOF
+    local is_cython_used=$(python3 <<EOF
```
Comment (ChaiBapchya) [inline ci/docker/runtime_functions.sh]: does this work with incomplete var?

Comment (leezu) [reply ci/docker/runtime_functions.sh]: What do you mean? `check_cython` is only called at one place and I modified that place to not pass the `python_ver`.

Comment (ChaiBapchya) [reply ci/docker/runtime_functions.sh]: I thought it should be 
```suggestion
    local is_cython_used=$(python3) <<EOF
```

Comment (leezu) [reply ci/docker/runtime_functions.sh]: I think the closing brace comes in line 52.

**Pull request merged**